server/src/model.ts

```diff
@@ -1502,6 +1556,114 @@ export async function getAssignmentDataFromCode(
   return { assignmentFound: true, assignment };
 }
 
+export async function searchPublishedLibraryContent({
```

Comment:
The only reason for a separate query from `searchSharedContent` is to have separate paginations. (There's also a small difference in that contentShares are ignored for the library, but it wouldn't have an effect if they were included.)

I believe, we could combine `searchPublishedLibraryContent` and `searchSharedContent` into a single query using the ROW_NUMBER MySQL window function as illustrated in [this example](https://stackoverflow.com/a/30269273). Select a field like `content.ownerId = ${libraryId} As inLibrary`, and then PARTITION based on that? As if this SQL statement wasn't sufficiently complicated already...

Reply:
That solution would only work for the first page, though. So maybe it's not worth it. At the very least, we could combine these APIs into a single function with a parameter that just changes the WHERE clause.